### DISCUSSION TITLE
Preferred way to produce metrics?

### DISCUSSION THREAD

**Discussion by bonndan** [NovatecConsulting/novatec-service-dependency-graph-panel#62] [Q&A]
What is your preferred way to produce the appropriate metrics for the panel (e.g. source... target...)? Do you have some kind of  library or the like to plug into services?

**Comment by mariusoe** [comment]
Hi,

actually, there is not really a preferred way to produce the required metrics. Since the graph operates independently of the Grafana data source used, the origin of the data does not matter as long as all the required tags are present.

In case your application landscape is based on Java, you could use our Java agent inspectit-ocelot (https://github.com/inspectIT/inspectit-ocelot/) for collecting the metrics. The agent supports collecting the metrics (required by the service dependency graph panel) out of the box and can be integrated quite easily.
